### PR TITLE
[#391] Add AAI Token to the user model

### PR DESCRIPTION
**Pull request by wujuu**

### CODE REVIEW

File: recommender/api/schemas/database_dump/user.py [hunk 1]
```diff
@@ -13,6 +13,12 @@
             description="The unique identifier of a the user",
             example=1234,
         ),
+        "aai_token": fields.String(
+            required=True,
+            title="AAI Token",
+            description="AAI token of the user",
+            example="abcdefghijklmnop",
```
Comment (Michal-Kolomanski) [inline recommender/api/schemas/database_dump/user.py:20]: This example is misleading and may cause confusion.
An example of an uid can be: 64-characters@egi.eu
General structure: [id-hex]@[sso-provider]

File: recommender/api/schemas/database_dump/user.py [hunk 2]
```diff
@@ -13,6 +13,12 @@
             description="The unique identifier of a the user",
             example=1234,
         ),
+        "aai_token": fields.String(
```
Comment (Michal-Kolomanski) [inline recommender/api/schemas/database_dump/user.py:16]: I believe it should be named "uid" not aai_token - but that's my bad. The title of the ticket was misleading. Sorry for that. Please change that everywhere

**Review by Michal-Kolomanski — CHANGES_REQUESTED** [review]
Unfortunately, the scope of the ticket is larger. "Adjust models and code regarding introducing the AAI token", please also change our code in such a way that we will be using **uid** or **user_id** for identifying users. There is also open question: shouldn't we use only **uid** for identifying users?

**Comment by wujuu** [comment]
I don’t understand why we cannot use user-id for the existing models and only use uid (aai-token) for the new engines that require this new identification scheme

**Comment by Michal-Kolomanski** [comment]
> I don’t understand why we cannot just use user-id (like we’ve been doing this whole time) for the existing models and only use uid (aai-token) for the new engines that require this new identification scheme

We have a complete understanding here, that's how it should be done. What I only suggested that still should be done was the ability to use the uid (in endpoint / recommendations) which implies: 
1) correctly identify the user based on his uid in recommendations pipelines
2) receive relevant recommendations for the user

But I can agree that this can be the scope of the next ticket. If you want, we can merge this ticket (after aai_token -> uid) and work on the above feature in the next ticket

**Comment by wujuu** [comment]
Made a change to intake aai_token or user_id as a user identification type. Only tested through unit tests for now.
and BTW:
Can we actually not rename the field? user_id and uid seem way to similar in my opinion

File: recommender/api/schemas/database_dump/user.py (hunk 2 above)
Comment (Michal-Kolomanski) [inline recommender/api/schemas/database_dump/user.py:16]: **aai_uid** in my opinion will be better

File: recommender/api/schemas/database_dump/user.py (hunk 1 above)
Comment (Michal-Kolomanski) [inline recommender/api/schemas/database_dump/user.py:20]: An example may be better. Look [recommender/api/schemas/recommendation.py](https://github.com/cyfronet-fid/recommender-system/pull/394/files#diff-a395fd459f377dc7e7a1575bf337d5e2bb9a7eafbbd5e77070ce7d9b802fbbe1)

            example="zooehewplbfxqtnxpspzsgrmeadoynbygyvisuvzst"
            + "dgenzxklzuwfdjndtzitiz@egi.eu",

File: recommender/api/schemas/recommendation.py [hunk 3]
```diff
@@ -52,6 +52,14 @@
             description="The unique identifier of the logged user. ",
             example=1,
         ),
+        "aai_token": fields.String(
```
Comment (Michal-Kolomanski) [inline recommender/api/schemas/recommendation.py:55]: aai_uid

File: recommender/api/schemas/user_action.py [hunk 4]
```diff
@@ -117,6 +117,14 @@
             description="The unique identifier of the logged user.",
             example=1234,
         ),
+        "aai_token": fields.String(
```
Comment (Michal-Kolomanski) [inline recommender/api/schemas/user_action.py:120]: aai_uid

File: recommender/engines/base/base_inference_component.py [hunk 5]
```diff
@@ -85,20 +86,26 @@ def _load_models(self) -> None:
     @staticmethod
     def _get_user(context: Dict[str, Any]) -> User:
         """
-        Get the user from the context.
+        Get the user from the context using either a user_id or an aai_token.
```
Comment (Michal-Kolomanski) [inline recommender/engines/base/base_inference_component.py:89]: aai_uid

**Review by Michal-Kolomanski — CHANGES_REQUESTED** [review]
LGTM but please change aai_token to aai_uid or present some more arguments for the name aai_token

**Comment by wujuu** [comment]
Ok changes the `aai_token` to `aai_uid`

File: recommender/services/deserializer.py [hunk 6]
```diff
@@ -78,8 +80,12 @@ def deserialize_user_action(cls, json_dict):
         action_data = json_dict.get("action", {})
         action = Action(**action_data)
 
+        user_id, aai_uid = json_dict.get("user_id"), json_dict.get("aai_uid")
+        if aai_uid and not user_id:
```
Comment (Michal-Kolomanski) [inline recommender/services/deserializer.py:84]: Assumption that if aai_token is passed, user_id is not. Just thinking loudly. No action is required.
So in the MP we cannot pass also aai_token

**Review by Michal-Kolomanski — APPROVED** [review]
LGTM :clap:

**Comment by Michal-Kolomanski** [comment]
1 thing only - please add changelog

**Pull request merged**